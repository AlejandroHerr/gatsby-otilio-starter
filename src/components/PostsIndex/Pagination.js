// @flow
import React from 'react';
import Link from 'gatsby-link';
import classnames from 'classnames';

import type { PaginationType } from '../../types/postsIndex';

import ByteIcon from '../icons/ByteIcon';

import styles from './PostsIndex.module.scss';

const Pagination = ({ page, pageCount, pathPrefix }: PaginationType) => (
  <nav className={styles.posts_index_pagination}>
    <div className={styles.posts_index_pagination__inner}>
      {page > 1 &&
        <Link
          className={classnames(
            styles.posts_index_pagination__link,
            styles.posts_index_pagination__next,
          )}
          to={page > 2 ? `${pathPrefix}/${page - 1}` : '/'}
        >
          <span className={styles.posts_index_pagination__label}><ByteIcon icon="chevron_left" /> Newer Posts</span>
        </Link>
      }
      <span className={styles.posts_index_pagination__info}>{`Page ${page} of ${pageCount}`}</span>
      {page < pageCount &&
        <Link
          className={classnames(
            styles.posts_index_pagination__link,
            styles.posts_index_pagination__prev,
          )}
          to={`${pathPrefix}/${page + 1}`}
        >
          <span className={styles.posts_index_pagination__label}>Older Posts <ByteIcon icon="chevron_right" /></span>
        </Link>
      }
      <div className={styles.posts_index_pagination__clear} />
      <div className={styles.posts_index_pagination__separator} />
    </div>
  </nav>
);

export default Pagination;
