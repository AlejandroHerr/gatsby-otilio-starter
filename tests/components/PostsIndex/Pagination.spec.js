import React from 'react';
import { shallow } from 'enzyme';

import Link from 'gatsby-link';

import Pagination from '../../../src/components/PostsIndex/Pagination';

const defaultProps = {
  page: 1,
  pageCount: 1,
  pathPrefix: 'prefix',
};

const setup = (props = {}) => {
  const pagination = shallow(<Pagination {...defaultProps} {...props} />);

  return {
    pagination,
    link: pagination.find(Link),
    info: pagination.find('.posts_index_pagination__info'),
  };
};

describe('PostsIndex', () => {
  describe('Pagination', () => {
    it('should render a pagination for one page only', () => {
      const { pagination, link, info } = setup();

      expect(link).toHaveLength(0);
      expect(info.text()).toBe(`Page ${defaultProps.page} of ${defaultProps.pageCount}`);
      expect(pagination).toMatchSnapshot();
    });
    it('should render pagination for the first page', () => {
      const props = {
        ...defaultProps,
        pageCount: 2,
      };
      const { pagination, link, info } = setup(props);

      expect(link).toHaveLength(1);
      expect(link.first().prop('to')).toBe(`${props.pathPrefix}/page/${props.page + 1}`);
      expect(info.text()).toBe(`Page ${props.page} of ${props.pageCount}`);
      expect(pagination).toMatchSnapshot();
    });
    it('should render pagination for middle page', () => {
      const props = {
        ...defaultProps,
        page: 2,
        pageCount: 3,
      };
      const { pagination, link, info } = setup(props);

      expect(link).toHaveLength(2);
      expect(link.first().prop('to')).toBe(props.pathPrefix);
      expect(info.text()).toBe(`Page ${props.page} of ${props.pageCount}`);
      expect(link.last().prop('to')).toBe(`${props.pathPrefix}/page/${props.page + 1}`);
      expect(pagination).toMatchSnapshot();
    });
    it('should render pagination for the last page', () => {
      const props = {
        ...defaultProps,
        page: 3,
        pageCount: 3,
      };
      const { pagination, link, info } = setup(props);

      expect(link).toHaveLength(1);
      expect(link.first().prop('to')).toBe(`${props.pathPrefix}/page/${props.page - 1}`);
      expect(info.text()).toBe(`Page ${props.page} of ${props.pageCount}`);
      expect(pagination).toMatchSnapshot();
    });
  });
});
