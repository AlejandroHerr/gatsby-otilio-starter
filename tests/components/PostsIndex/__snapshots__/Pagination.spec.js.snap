// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsIndex Pagination should render a pagination for one page only 1`] = `
<nav
  className="posts_index_pagination"
>
  <div
    className="posts_index_pagination__inner"
  >
    <span
      className="posts_index_pagination__info"
    >
      Page 1 of 1
    </span>
    <div
      className="posts_index_pagination__clear"
    />
    <div
      className="posts_index_pagination__separator"
    />
  </div>
</nav>
`;

exports[`PostsIndex Pagination should render pagination for middle page 1`] = `
<nav
  className="posts_index_pagination"
>
  <div
    className="posts_index_pagination__inner"
  >
    <GatsbyLink
      className="posts_index_pagination__link posts_index_pagination__next"
      to="prefix"
    >
      <span
        className="posts_index_pagination__label"
      >
        <ByteIcon
          bold={false}
          className=""
          icon="chevron_left"
        />
         Newer Posts
      </span>
    </GatsbyLink>
    <span
      className="posts_index_pagination__info"
    >
      Page 2 of 3
    </span>
    <GatsbyLink
      className="posts_index_pagination__link posts_index_pagination__prev"
      to="prefix/page/3"
    >
      <span
        className="posts_index_pagination__label"
      >
        Older Posts 
        <ByteIcon
          bold={false}
          className=""
          icon="chevron_right"
        />
      </span>
    </GatsbyLink>
    <div
      className="posts_index_pagination__clear"
    />
    <div
      className="posts_index_pagination__separator"
    />
  </div>
</nav>
`;

exports[`PostsIndex Pagination should render pagination for the first page 1`] = `
<nav
  className="posts_index_pagination"
>
  <div
    className="posts_index_pagination__inner"
  >
    <span
      className="posts_index_pagination__info"
    >
      Page 1 of 2
    </span>
    <GatsbyLink
      className="posts_index_pagination__link posts_index_pagination__prev"
      to="prefix/page/2"
    >
      <span
        className="posts_index_pagination__label"
      >
        Older Posts 
        <ByteIcon
          bold={false}
          className=""
          icon="chevron_right"
        />
      </span>
    </GatsbyLink>
    <div
      className="posts_index_pagination__clear"
    />
    <div
      className="posts_index_pagination__separator"
    />
  </div>
</nav>
`;

exports[`PostsIndex Pagination should render pagination for the last page 1`] = `
<nav
  className="posts_index_pagination"
>
  <div
    className="posts_index_pagination__inner"
  >
    <GatsbyLink
      className="posts_index_pagination__link posts_index_pagination__next"
      to="prefix/page/2"
    >
      <span
        className="posts_index_pagination__label"
      >
        <ByteIcon
          bold={false}
          className=""
          icon="chevron_left"
        />
         Newer Posts
      </span>
    </GatsbyLink>
    <span
      className="posts_index_pagination__info"
    >
      Page 3 of 3
    </span>
    <div
      className="posts_index_pagination__clear"
    />
    <div
      className="posts_index_pagination__separator"
    />
  </div>
</nav>
`;
